server/testsupport/curiousAssessmentsTestDataBuilder.ts

```diff
@@ -0,0 +1,469 @@
+  LearnerLddInfoExternalV1DTO,
+  LearnerAssessmentV2DTO,
+  ExternalAssessmentsDTO,
+  LearnerAssessmentsDTO,
+  LearnerAssessmentsFunctionalSkillsDTO,
+  LearnerAssessmentsAlnDTO,
+} from 'curiousApiClient'
+
+const anAllAssessmentDTO = (
+  options: {
+    v1Assessments?: Array<LearnerLatestAssessmentV1DTO>
+    v2Assessments?: LearnerAssessmentV2DTO
+  } = {
+    v1Assessments: [aLearnerLatestAssessmentV1DTO()],
+    v2Assessments: aLearnerAssessmentV2DTO(),
+  },
+): AllAssessmentDTO => ({
+  v1: options.v1Assessments,
+  v2: options.v2Assessments,
+})
+
+const aLearnerLatestAssessmentV1DTO = (
+  options: {
+    prisonNumber?: string
+    qualifications?: Array<LearnerAssessmentV1DTO>
+    lddAssessments?: Array<LearnerLddInfoExternalV1DTO>
+  } = {
+    prisonNumber: 'A1234BC',
+    qualifications: [aLearnerAssessmentV1DTO()],
+    lddAssessments: [aLearnerLddInfoExternalV1DTO()],
+  },
+): LearnerLatestAssessmentV1DTO => ({
+  prn: options.prisonNumber,
+  qualifications: options.qualifications,
+  ldd: options.lddAssessments,
+})
+
+const aLearnerAssessmentV1DTO = (
+  options: {
+    prisonId?: string
+    prisonName?: string
+    qualificationType?: 'English' | 'Maths' | 'Digital Literacy'
+    qualificationGrade?: string
+    assessmentDate?: string
+  } = {
+    prisonId: 'MDI',
+    prisonName: 'MOORLAND (HMP & YOI)',
+    qualificationType: 'English',
+    qualificationGrade: 'Level 1',
+    assessmentDate: '2012-02-16',
+  },
+): LearnerAssessmentV1DTO => ({
+  establishmentId: options.prisonId,
+  establishmentName: options.prisonName,
+  qualification: {
+    qualificationType: options.qualificationType,
+    qualificationGrade: options.qualificationGrade,
+    assessmentDate: options.assessmentDate,
+  },
+})
+
+const aLearnerLddInfoExternalV1DTO = (
+  options: {
+    prisonId?: string
+    prisonName?: string
+    lddPrimaryName?: string
+    lddSecondaryNames?: string[]
+    inDepthAssessmentDate?: string
+    rapidAssessmentDate?: string
+  } = {
+    prisonId: 'MDI',
+    prisonName: 'MOORLAND (HMP & YOI)',
+    lddPrimaryName: 'Visual impairment',
+    lddSecondaryNames: ['Hearing impairment', 'Mental health difficulty', 'Social and emotional difficulties'],
+    inDepthAssessmentDate: '2012-02-16',
+    rapidAssessmentDate: '2012-02-16',
+  },
+): LearnerLddInfoExternalV1DTO => ({
+  establishmentId: options.prisonId,
+  establishmentName: options.prisonName,
+  lddPrimaryName: options.lddPrimaryName,
+  lddSecondaryNames: options.lddSecondaryNames,
+  inDepthAssessmentDate: options.inDepthAssessmentDate,
+  rapidAssessmentDate: options.rapidAssessmentDate,
+})
+
+const aLearnerAssessmentV2DTO = (
+  options: {
+    prisonNumber?: string
+    assessments?: ExternalAssessmentsDTO
+  } = {
+    prisonNumber: 'A1234BC',
+    assessments: anExternalAssessmentsDTO(),
+  },
+): LearnerAssessmentV2DTO => ({
+  prn: options.prisonNumber,
+  assessments: options.assessments,
+})
+
+const anExternalAssessmentsDTO = (
```

Comment:
This is the core builder for the `v2` assessments.
It contains an array of each of ALN, Digital Functional Skills, Maths Functional Skills, English Functional Skills, ESOL and Reading assessments.

There are some subtly different types here ...
The ALN assessments are of type `LearnerAssessmentsAlnDTO`
The Functional Skills assessments (digital, maths and English) are all of type `LearnerAssessmentsFunctionalSkillsDTO` BUT their field values are different! See the relevant builders below
And ESOL and Reading are of type `LearnerAssessmentsDTO`